From 5e65bdabc7b16e2bfe4f1757cba3485b0a043daa Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Tue, 22 Nov 2022 10:29:51 +0100
Subject: [PATCH 126/172] Remove pointer indirection where not needed

This functions do not need to use a pointer to a pointer.
---
 etc.c      |   30 +++++++++++++++---------------
 file.c     |    2 +-
 istream.c  |    6 +++---
 mimehead.c |   38 +++++++++++++++++++-------------------
 proto.h    |   16 ++++++++--------
 5 files changed, 46 insertions(+), 46 deletions(-)

--- a/etc.c
+++ b/etc.c
@@ -120,41 +120,41 @@ currentLineSkip(Buffer *buf, Line *line,
 #define MAX_CMD_LEN 128
 
 int
-gethtmlcmd(char **s)
+gethtmlcmd(char *s)
 {
     extern Hash_si tagtable;
     char cmdstr[MAX_CMD_LEN];
     char *p = cmdstr;
-    char *save = *s;
+    char *save = s;
     int cmd;
 
-    (*s)++;
+    s++;
     /* first character */
-    if (IS_ALNUM(**s) || **s == '_' || **s == '/') {
-	*(p++) = TOLOWER(**s);
-	(*s)++;
+    if (IS_ALNUM(*s) || *s == '_' || *s == '/') {
+	*(p++) = TOLOWER(*s);
+	s++;
     }
     else
 	return HTML_UNKNOWN;
     if (p[-1] == '/')
-	SKIP_BLANKS(*s);
-    while ((IS_ALNUM(**s) || **s == '_') && p - cmdstr < MAX_CMD_LEN) {
-	*(p++) = TOLOWER(**s);
-	(*s)++;
+	SKIP_BLANKS(s);
+    while ((IS_ALNUM(*s) || *s == '_') && p - cmdstr < MAX_CMD_LEN) {
+	*(p++) = TOLOWER(*s);
+	s++;
     }
     if (p - cmdstr == MAX_CMD_LEN) {
 	/* buffer overflow: perhaps caused by bad HTML source */
-	*s = save + 1;
+	s = save + 1;
 	return HTML_UNKNOWN;
     }
     *p = '\0';
 
     /* hash search */
     cmd = getHash_si(&tagtable, cmdstr, HTML_UNKNOWN);
-    while (**s && **s != '>')
-	(*s)++;
-    if (**s == '>')
-	(*s)++;
+    while (*s && *s != '>')
+	s++;
+    if (*s == '>')
+	s++;
     return cmd;
 }
 
--- a/file.c
+++ b/file.c
@@ -2669,7 +2669,7 @@ passthrough(struct readbuffer *obuf, cha
 	str_bak = str;
 	if (sloppy_parse_line(&str)) {
 	    char *q = str_bak;
-	    cmd = gethtmlcmd(&q);
+	    cmd = gethtmlcmd(q);
 	    if (back) {
 		struct link_stack *p;
 		for (p = link_stack; p; p = p->next) {
--- a/istream.c
+++ b/istream.c
@@ -702,11 +702,11 @@ ens_read(struct ens_handle *handle, char
 	growbuf_init_without_GC(&gbtmp);
 	p = handle->gb.ptr;
 	if (handle->encoding == ENC_QUOTE)
-	    decodeQP_to_growbuf(&gbtmp, &p);
+	    decodeQP_to_growbuf(&gbtmp, p);
 	else if (handle->encoding == ENC_BASE64)
-	    decodeB_to_growbuf(&gbtmp, &p);
+	    decodeB_to_growbuf(&gbtmp, p);
 	else if (handle->encoding == ENC_UUENCODE)
-	    decodeU_to_growbuf(&gbtmp, &p);
+	    decodeU_to_growbuf(&gbtmp, p);
 	growbuf_clear(&handle->gb);
 	handle->gb = gbtmp;
 	handle->pos = 0;
--- a/mimehead.c
+++ b/mimehead.c
@@ -62,7 +62,7 @@ ha2d(char x, char y)
 }
 
 Str
-decodeB(char **ww)
+decodeB(char *ww)
 {
     struct growbuf gb;
 
@@ -72,10 +72,10 @@ decodeB(char **ww)
 }
 
 void
-decodeB_to_growbuf(struct growbuf *gb, char **ww)
+decodeB_to_growbuf(struct growbuf *gb, char *ww)
 {
     unsigned char c[4];
-    char *wp = *ww;
+    char *wp = ww;
     char d[3];
     int i, n_pad;
 
@@ -118,15 +118,15 @@ decodeB_to_growbuf(struct growbuf *gb, c
 last:
     growbuf_reserve(gb, gb->length + 1);
     gb->ptr[gb->length] = '\0';
-    *ww = wp;
+    ww = wp;
     return;
 }
 
 void
-decodeU_to_growbuf(struct growbuf *gb, char **ww)
+decodeU_to_growbuf(struct growbuf *gb, char *ww)
 {
     unsigned char c1, c2;
-    char *w = *ww;
+    char *w = ww;
     int n, i;
 
     if (*w <= 0x20 || *w >= 0x60)
@@ -152,9 +152,9 @@ decodeU_to_growbuf(struct growbuf *gb, c
 
 /* RFC2047 (4.2. The "Q" encoding) */
 Str
-decodeQ(char **ww)
+decodeQ(char *ww)
 {
-    char *w = *ww;
+    char *w = ww;
     Str a = Strnew_size(strlen(w));
 
     for (; *w != '\0' && *w != '?'; w++) {
@@ -169,14 +169,14 @@ decodeQ(char **ww)
 	else
 	    Strcat_char(a, *w);
     }
-    *ww = w;
+    ww = w;
     return a;
 }
 
 void
-decodeQP_to_growbuf(struct growbuf *gb, char **ww)
+decodeQP_to_growbuf(struct growbuf *gb, char *ww)
 {
-    char *w = *ww;
+    char *w = ww;
 
     growbuf_reserve(gb, strlen(w) + 1);
     for (; *w != '\0'; w++) {
@@ -199,22 +199,22 @@ decodeQP_to_growbuf(struct growbuf *gb,
 	    gb->ptr[gb->length++] = *w;
     }
     gb->ptr[gb->length] = '\0';
-    *ww = w;
+    ww = w;
     return;
 }
 
 #ifdef USE_M17N
 Str
-decodeWord(char **ow, wc_ces * charset)
+decodeWord(char *ow, wc_ces * charset)
 #else
 Str
-decodeWord0(char **ow)
+decodeWord0(char *ow)
 #endif
 {
 #ifdef USE_M17N
     wc_ces c;
 #endif
-    char *p, *w = *ow;
+    char *p, *w = ow;
     char method;
     Str a = Strnew();
     Str tmp = Strnew();
@@ -244,10 +244,10 @@ decodeWord0(char **ow)
     p = w;
     switch (TOUPPER(method)) {
     case 'B':
-	a = decodeB(&w);
+	a = decodeB(w);
 	break;
     case 'Q':
-	a = decodeQ(&w);
+	a = decodeQ(w);
 	break;
     default:
 	goto convert_fail;
@@ -259,7 +259,7 @@ decodeWord0(char **ow)
 	if (*w == '=')
 	    w++;
     }
-    *ow = w;
+    ow = w;
 #ifdef USE_M17N
     *charset = c;
 #endif
@@ -295,7 +295,7 @@ decodeMIME0(Str orgstr)
 	    }
 	  nextEncodeWord:
 	    p = org;
-	    Strcat(cnv, decodeWord(&org, charset));
+	    Strcat(cnv, decodeWord(org, charset));
 	    if (org == p) {	/* Convert failure */
 		Strcat_charp(cnv, org);
 		return cnv;
--- a/proto.h
+++ b/proto.h
@@ -343,7 +343,7 @@ extern int columnPos(Line *line, int col
 extern int columnLen(Line *line, int column);
 extern Line *lineSkip(Buffer *buf, Line *line, int offset, int last);
 extern Line *currentLineSkip(Buffer *buf, Line *line, int offset, int last);
-extern int gethtmlcmd(char **s);
+extern int gethtmlcmd(char *s);
 extern Str checkType(Str s, Lineprop **oprop, Linecolor **ocolor);
 extern int calcPosition(char *l, Lineprop *pr, int len, int pos, int bpos,
 			int mode);
@@ -580,16 +580,16 @@ extern void shiftAnchorPosition(AnchorLi
 extern char *getAnchorText(Buffer *buf, AnchorList *al, Anchor *a);
 extern Buffer *link_list_panel(Buffer *buf);
 
-extern Str decodeB(char **ww);
-extern void decodeB_to_growbuf(struct growbuf *gb, char **ww);
-extern Str decodeQ(char **ww);
-extern void decodeQP_to_growbuf(struct growbuf *gb, char **ww);
-extern void decodeU_to_growbuf(struct growbuf *gb, char **ww);
+extern Str decodeB(char *ww);
+extern void decodeB_to_growbuf(struct growbuf *gb, char *ww);
+extern Str decodeQ(char *ww);
+extern void decodeQP_to_growbuf(struct growbuf *gb, char *ww);
+extern void decodeU_to_growbuf(struct growbuf *gb, char *ww);
 #ifdef USE_M17N
-extern Str decodeWord(char **ow, wc_ces * charset);
+extern Str decodeWord(char *ow, wc_ces * charset);
 extern Str decodeMIME(Str orgstr, wc_ces * charset);
 #else
-extern Str decodeWord0(char **ow);
+extern Str decodeWord0(char *ow);
 extern Str decodeMIME0(Str orgstr);
 #define decodeWord(ow,charset) decodeWord0(ow)
 #define decodeMIME(orgstr,charset) decodeMIME0(orgstr)
